src/util/event-to-promise.test.js

```diff
@@ -0,0 +1,144 @@
+/* eslint-env jest */
+
+import 'core-js/fn/object/entries' // shim Object.entries
+import pull from 'lodash/pull'
+
+import eventToPromise from './event-to-promise'
+
+class MockEvent {
+    constructor() {
+        this.listeners = []
+    }
+
+    addListener(listener) {
+        this.listeners.push(listener)
+    }
+
+    removeListener(listener) {
+        pull(this.listeners, listener)
+    }
+
+    trigger(...args) {
+        this.listeners.forEach(listener => listener.apply(null, args))
+    }
+}
+
+describe('eventToPromise', () => {
+    test('should return a promise', () => {
+        const promise = eventToPromise({})
+        expect(promise).toBeInstanceOf(Promise)
+    })
+
+    test('should listen and unlisten to the events', async () => {
+        // We try both passing multiple events (for resolveOpts) and a single event (for rejectOpts).
+        const resolveOpts = [
+            { event: new MockEvent() },
+            { event: new MockEvent() },
+        ]
+        const rejectOpts = { event: new MockEvent() }
+        eventToPromise({
+            resolve: resolveOpts,
+            reject: rejectOpts,
+        })
+
+        // We use a bogus await statement to let any resolved promises invoke their callbacks.
+        // XXX Not sure if we can rely on this in every ES implementation.
+        await null
```

Comment:
How does `await null` work? Wouldn't it just continue on immediately? Or is it similar to passing `0` to `setTimeout`, putting the subsequent code at the end of the execution queue?

Reply:
[Here](https://github.com/WebMemex/webmemex-extension/pull/124#pullrequestreview-62017180) is some context, I hope this provides some help regarding the clarification of the use of ```await null```. I am still searching for a better alternative. Hopefully the upgrade in jest might just resolve the issue.